pkg/client/userd/trafficmgr/mount.go

```diff
@@ -34,6 +52,41 @@ func (ic *intercept) startMount(ctx context.Context, podWG *sync.WaitGroup) {
 	}
 }
 
+type ftpMounter struct {
+	client fs.FTPClient
+}
+
+func (m *ftpMounter) start(_ context.Context, ic *intercept) error {
+	// The FTPClient and the NewHost must be controlled by the intercept context and not by the pod context that
+	// is passed as a parameter, because those services will survive pod changes.
+	ctx := ic.ctx
+	addr := netip.MustParseAddrPort(fmt.Sprintf("%s:%d", ic.PodIp, ic.FtpPort))
+	if m.client == nil {
+		dlog.Infof(ctx, "Mounting FTP file system for intercept %q (address %s) at %q", ic.Id, addr, ic.ClientMountPoint)
+		// FTPs remote mount is already relative to the agentconfig.ExportsMountPoint
+		rmp := strings.TrimPrefix(ic.MountPoint, agentconfig.ExportsMountPoint)
+
+		client, err := fs.NewFTPClient(ctx, addr, rmp, 5*time.Second)
+		if err != nil {
+			return err
+		}
+		m.client = client
+		host := fs.NewHost(m.client, ic.ClientMountPoint)
+		host.Start(ctx)
+		go func() {
+			// Stop when the intercept dies (but not when the pod changes)
+			<-ctx.Done()
+			defer dlog.Infof(ctx, "Unmounting FTP file system for intercept %q (address %s) at %q", ic.Id, addr, ic.ClientMountPoint)
+			host.Stop()
+		}()
+		return nil
+	}
```

Comment:
should we lock the client here?

Reply:
No, I don't think that's necessary. The `handleInterceptShapshot` method is dealing with intercepts sequentially, so there's no risk that the start method is called concurrently.